pkgs/os-specific/darwin/apple-source-releases/ICU/package.nix

```diff
@@ -0,0 +1,208 @@
+      ]
+      ++ lib.optionals withStatic [ (lib.enableFeature true "static") ];
+
+    nativeBuildInputs = [ python3 ];
+
+    enableParallelBuilding = true;
+
+    # Per the source-release makefile, these are enabled.
+    env.NIX_CFLAGS_COMPILE = toString [
+      "-DU_SHOW_CPLUSPLUS_API=1"
+      "-DU_SHOW_INTERNAL_API=1"
+    ];
+
+    passthru.tests.pkg-config = testers.testMetaPkgConfig finalAttrs.finalPackage;
+    passthru.buildRootOnly = mkWithAttrs buildRootOnlyAttrs;
+
+    meta = {
+      description = "Unicode and globalization support library with Apple customizations";
+      license = [ lib.licenses.icu ];
+      maintainers = lib.teams.darwin.members;
+      platforms = lib.platforms.darwin;
+      pkgConfigModules = [
+        "icu-i18n"
+        "icu-io"
+        "icu-uc"
+      ];
+    };
+  };
+
+  realAttrs = self: super: {
+    outputs = [
+      "out"
+      "dev"
+    ] ++ lib.optional withStatic "static";
+    outputBin = "dev";
+
+    postPatch = lib.optionalString self.finalPackage.doCheck ''
+      # Skip test for missing encodingSamples data.
+      substituteInPlace test/cintltst/ucsdetst.c \
+        --replace-fail "&TestMailFilterCSS" "NULL"
+
+      # Disable failing tests
+      substituteInPlace test/cintltst/cloctst.c \
+        --replace-fail 'TESTCASE(TestCanonicalForm);' ""
+
+      substituteInPlace test/intltest/rbbitst.cpp \
+        --replace-fail 'TESTCASE_AUTO(TestExternalBreakEngineWithFakeYue);' ""
+
+      # Otherwise `make install` is broken.
+      substituteInPlace Makefile.in \
+        --replace-fail '$(top_srcdir)/../LICENSE' "$NIX_BUILD_TOP/source/icu/LICENSE"
+      substituteInPlace config/dist-data.sh \
+        --replace-fail "\''${top_srcdir}/../LICENSE" "$NIX_BUILD_TOP/source/icu/LICENSE"
+    '';
+
+    # remove dependency on bootstrap-tools in early stdenv build
+    postInstall =
+      lib.optionalString withStatic ''
+        mkdir -p $static/lib
+        mv -v lib/*.a $static/lib
+      ''
+      + lib.optionalString stdenv.hostPlatform.isDarwin ''
+        sed -i 's/INSTALL_CMD=.*install/INSTALL_CMD=install/' $out/lib/icu/*/pkgdata.inc
+      ''
+      + (
+        let
+          replacements = [
+            {
+              from = "\${prefix}/include";
+              to = "${placeholder "dev"}/include";
+            } # --cppflags-searchpath
+            {
+              from = "\${pkglibdir}/Makefile.inc";
+              to = "${placeholder "dev"}/lib/icu/Makefile.inc";
+            } # --incfile
+            {
+              from = "\${pkglibdir}/pkgdata.inc";
+              to = "${placeholder "dev"}/lib/icu/pkgdata.inc";
+            } # --incpkgdatafile
+          ];
+        in
+        ''
+          rm $out/share/icu/*/install-sh $out/share/icu/*/mkinstalldirs # Avoid having a runtime dependency on bash
+
+          substituteInPlace "$dev/bin/icu-config" \
+            ${lib.concatMapStringsSep " " (r: "--replace '${r.from}' '${r.to}'") replacements}
```

Comment:
I made the change, but I need to complete a bootstrap to confirm it didn’t break the ICU build.

Reply:
This was more about record keeping, if a rebuild takes too long just skip it for now.

Reply:
The only really lengthy build is second-stage LLVM (due to system_cmds). Since I changed actual code, it would be prudent to make sure things still actually built.

Reply:
Skip the change is what I meant. Not the rebuild (though yes the rebuild through not causing it in the first place).

Reply:
ICU built, so I think this is fine.